### DISCUSSION TITLE
Concurrency limits using AWS Lambda

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#13821] [General]
Hi,

I am looking at different database options for a platform leveraging AWS Lambda. We have a high degree of concurrency, probably over 1000 concurrent lambdas and are currently running into connection limits with MongoDB.

My question is what considerations should be be aware of if we were to use RavenDB in a similar setup, what issues, if any might we face spinning up 100's of document stores concurrently, using them briefly while the lambda is running and then disposing of them.

Thanks

**Comment by ayende** [comment]
There shouldn't be any issues in this regard.

In Lambda, be _sure_ that you are storing the `DocumentStore` in a global manner, no per lambda invocation.
The Lambda infrastructure will process multiple requests in the same instance, so keeping the same `DocumentStore` is a big help.

In terms of number of connections, we have tested RavenDB with 5000+ connections (and tens of thousands of requests per second) with no major issues.

**Reply by nickchampion** [reply]
Thanks @ayende can I ask, the 5000+ connections was that on a single node, a 3 node cluster? commodity hardware? can you provide some context to these numbers please

**Reply by ayende** [reply]
You can see some numbers here, with specs:

https://ravendb.net/whitepapers/couchbase-vs-ravendb-performance-at-rakuten-kobo

5000+ connections may depend on your dynamic port range if running from a single location, btw. 

If you are talking about multiple clients and the total number of connections, a single node running on Raspberry Pi can handle > 10K req/sec for load by id.  IIRC, we did that with 512 connections. 

Just tried running 256 connections on the live tests (remote machine, ~600ms latency to that), and I got this:

![image](https://user-images.githubusercontent.com/116915/158557473-81f7d5da-1849-4852-a134-fe2a85b578d9.png)

5000+ on a single node isn't something that requires huge hardware.

**Reply by nickchampion** [reply]
Thanks @ayende this is perfect
